### DISCUSSION TITLE
How does one return from a mark

### DISCUSSION THREAD

**Discussion by tororutsu** [nvim-neorg/neorg#129] [Q&A] [answered]
After finally learning how to follow a mark I am lost on going back? Is there a keybinding for that or a preferred method. and is there a way to possibly follow a mark in a split?

**Comment by mrossinek** [comment, marked as answer]
Hi @rjahrj! Thanks for bringing this to our attention! I had previously noticed, too, that the `jumplist` is not being leveraged while following links but I had not gotten around to look into it.
That being said, I just pushed a fix for this to the `unstable` branch which allows you to use `<C-o>` and `<C-i>` after following links just like you would normally navigate your previous jumps :+1: 

As per following links in splits that is a very good suggestion! I will open an issue to track this idea right away :+1: 
EDIT: Link to issue: #131

**Reply by tororutsu** [reply]
Wow man thanks this worked! I learned 2 thinks from this; Packer makes it very easy to switch branches and marks are incredible. This is my first time trying to seriously invest in a plugin. I am a newbie but I am learning lol. I

I also have a question related to marking. Can I mark a file in a subdir.
 
> notes/ 
    index.norg  
    programming/
    index.norg

**Reply by mrossinek** [reply]
> Can I mark a file in a subdir.

I'm not sure I fully understand the question. If you want to know how to link to a location in a file in a subdirectory you can do:
```norg
[my awesome link](:programming/index:#my awesome location)
```

**Comment by filchristou** [comment]
> After finally learning how to follow a mark

Well, I am still struggling with this. Can you please provide how this can be done?

For example in the case of the documentation, how do you follow the link?
```
  $name marker1
  | My Special Marker

  $comment
  Reference the marker with its custom name

  I have a link to my marker right [here](#marker1).

```

**Reply by mrossinek** [reply]
If you have the default keybindings enabled in your config like so:
```lua
        ["core.keybinds"] = {
            config = {
                default_keybinds = true,
            }
        },
```
Then you need to press `<Cr>` (Enter) while your cursor is on top of the link text. After doing so you will be jumped to the link target. You can then jump back and forth using the standard vim keybindings for walking the jumplist (`:help CTRL-O` and `:help CTRL-I`).

If you want to change the default keybinding you can find out how to do this on [this wiki page](https://github.com/nvim-neorg/neorg/wiki/User-Keybinds#quickstart). The "keybind megalist" at the end of that page is slightly out-of-date right now so here is the target which you need to link to:
```lua
        { "<CR>", "core.norg.esupports.goto_link" },
```
